src/cowboy_telemetry_h.erl

```diff
@@ -0,0 +1,57 @@
+-module(cowboy_telemetry_h).
+-behavior(cowboy_stream).
+
+-export([init/3]).
+-export([data/4]).
+-export([info/3]).
+-export([terminate/3]).
+-export([early_error/5]).
+
+init(StreamID, Req, Opts) ->
+  SystemTime = erlang:system_time(),
+  StartTime = erlang:monotonic_time(),
+  telemetry:execute(
+    [cowboy, request, start],
+    #{system_time => SystemTime},
+    #{stream_id => StreamID, req => Req}
+  ),
+  {Commands, Next} = cowboy_stream:init(StreamID, Req, Opts),
+  {Commands, [Next | StartTime]}.
+
+data(StreamID, IsFin, Data, [Next0 | StartTime]) ->
+  {Commands, Next} = cowboy_stream:data(StreamID, IsFin, Data, Next0),
+  {Commands, [Next | StartTime]}.
+
+info(StreamID, Info, [Next0 | StartTime]) ->
+  EndTime = erlang:monotonic_time(),
+  case Info of
+    {response, _, _, _} = Response ->
+      telemetry:execute(
+        [cowboy, request, stop],
+        #{duration => EndTime - StartTime},
+        #{stream_id => StreamID, response => Response}
+      );
+    {'EXIT', _, Reason} when Reason /= normal ->
```

Comment:
`Reason /= normal` means no exception nor stop events will be emitted. I am not sure that's the way we want to go. Another other ideas?

Reply:
I'm actually noticing that other handlers (like `cowboy_metrics_h`) respond to the `Commands` rather than the `Info`.. And I've found that an `error_response` tuple is part of the commands returned alongside an exit - it's shaped just like the `response` tuple. 

`Commands`:
* request process completes:
  * `[{response, _, _, _}]`
* request process fails:
  * `[{error_response, _, _, _}, _log_command, {internal_error, {EXIT, pid, reason}}]`

Now I'm thinking of reworking this so that the `exception` event is all based on `Commands`. It seems better to respond based on the commands cowboy gives rather than trying to pre-parse the info.

I'll push up a change for that soon...